modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -350,6 +365,14 @@ object Json {
     final def mapString(f: String => String): Json = this
     final def mapArray(f: Vector[Json] => Vector[Json]): Json = this
     final def mapObject(f: JsonObject => JsonObject): Json = this
+
+    final def traverseBoolean[F[_]](f: Boolean => F[Json])(implicit F: Applicative[F]): F[Json] =
+      F.map(f(value))(identity)
```

Comment:
I think we might be overly complicating the traverse methods here that are not just pure.  Here is my suggestion:

```suggestion
    final def traverseBoolean[F[_]](f: Boolean => F[Json])(implicit F: Applicative[F]): F[Json] = f(value)
```

This change would need to be applied to each of there other sum types here.

Reply:
Pushed a fix for this.